index.js

```diff
@@ -1,38 +1,16 @@
-function RandomCat() { }
-
-RandomCat.prototype.get = function (options) {
-  options = options || {};
-
-  if (options.dummyText) {
-    options.dummyText = escape(options.dummyText);
-  }
-
-  var randomSizes = [50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600];
-
-  var urlParts = [
-      options.gray ? 'g' : false
-    , options.width || randomSizes[Math.ceil(Math.random() * 12)]
-    , options.height || randomSizes[Math.ceil(Math.random() * 12)]
-    , options.category || 'cats'
-    , options.imageIndex
-    , options.dummyText
-  ];
-
-  var length = urlParts.length;
-  var protocol = options.protocol || 'http';
-  var baseUrl = options.baseUrl || 'lorempixel.com';
-  var lastSlash = baseUrl.lastIndexOf('/');
-  baseUrl = lastSlash === baseUrl.length - 1
-    ? baseUrl.substr(0, lastSlash)
-    : baseUrl;
-
-  for (var i=0; i<length; i++) {
-    if (!!!urlParts[i]) {
-      urlParts.splice(i, 1);
-    }
-  }
-
-  return protocol + '://' + baseUrl + '/' + urlParts.join('/');
+const RANDOM_SIZES = [50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600];
+
+class RandomCat {
+	constructor(options = {}) {
+		this.width = options.width || RANDOM_SIZES[Math.ceil(Math.random() * 12)];
+		this.height = options.height || RANDOM_SIZES[Math.ceil(Math.random() * 12)];
+		this.imageIndex = options.imageIndex || 0;
+		this.message = options.message || '';
+	}
+
+	getURL() {
+		return `https://lorempixel.com/${this.width}/${this.height}/cats/${this.imageIndex}/${this.message}`;
```

Comment:
Base url should also not be hard coded, should be configurable as before.  Note:  This would be a good candidate for creating a base options object and using object spread w/the options passed in.